src/main/java/me/recursiveg/Mc122477Fix.java

```diff
@@ -22,6 +23,7 @@ public void onChatOpen(GuiOpenEvent ev) {
 
     @SubscribeEvent
     public void onChatTyped(GuiScreenEvent.KeyboardCharTypedEvent.Pre ev) {
-        if (System.currentTimeMillis() - lastChatOpenTime < 50) ev.setCanceled(true);
+        if (System.currentTimeMillis() < lastChatOpenTime + (long)(Minecraft.getInstance().getTickLength() * 100))
```

Comment:
It seems the `getTickLength()` function is horribly named, and it actually returns `this.timer.elapsedPartialTicks`, which seems to me don't have anything to do with real-world time length.